Add display items to bestsellers report

missoxd opened the pull request:
<!---
    Thank you for contributing to Magento.
    To help us process this pull request we recommend that you add the following information:
     - Summary of the pull request,
     - Issue(s) related to the changes made,
     - Manual testing scenarios
    Fields marked with (*) are required. Please don't remove the template.
-->

<!--- Please provide a general summary of the Pull Request in the Title above -->

### Description (*)
<!---
    Please provide a description of the changes proposed in the pull request.
    Letting us know what has changed and why it needed changing will help us validate this pull request.
-->
As of now Magento Bestsellers report only shows 5 lines, this values is hardcoded into the code. The idea here is to create a new "Display items" field in the Bestsellers report form, this way the admin user can select if he wants to see more than 5 rows. It was used pre-defined values (5, 10, 20, 50), to prevent the admin user to accidentally request all rows and have some performance issues.

### Fixed Issues (if relevant)
<!---
    If relevant, please provide a list of fixed issues in the format magento/magento2#<issue_number>.
    There could be 1 or more issues linked here and it will help us find some more information about the reasoning behind this change.
-->
1. magento/magento2#21361: Bestsellers collection is using ratingLimit as sql select limit

### Manual testing scenarios (*)
<!---
    Please provide a set of unambiguous steps to test the proposed code change.
    Giving us manual testing scenarios will help with the processing and validation process.
-->
1. Have a Magento installation set up
2. Have completed orders with more than 5 different products
3. Log in into the admin panel
4. Go to Reports > Bestsellers
5. Fill in from and to date (needs to encapsulate the orders)
6. Select a Display items value
7. Click on Show report
8. The report needs to show the exact quantity selected at the Display items field, or all the rows if the selected value is greater than the number of bestsellers products

### Observations
- I didn't write any automated tests for this feature yet, would be nice to have some sort of help in this part
- In this Magento\Sales\Block\Adminhtml\Report\Bestsellers\Filter\Form i wasn't sure if i needed to use `@api` and `@since`

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

missoxd committed:
magento/magento2#21361: Bestsellers collection is using ratingLimit sql ...

- add display items field to bestsellers report

magento-cicd2 commented:
[![CLA assistant check](https://cla.dev.magento.com/pull/badge/not_signed)](https://cla.dev.magento.com/magento/magento2?pullRequest=22851) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=22851) before we can accept your contribution.<br/>

missoxd committed:
magento/magento2#21361: Bestsellers collection is using ratingLimit sql

- remove braces from inheritdoc
- remove api and since annotation from new class

swnsma commented:
Hi @missoxd 
Thank you for your contribution.
As you PR provide functional changes, I need to clarify this question with Product Owner of Magento Reports. It could take some time.

missoxd commented:
Hi @swnsma
Thank you for the answer.
No problem at all.

swnsma commented on app/code/Magento/Sales/Model/ResourceModel/Report/Bestsellers/Collection.php:
```diff
@@ -56,6 +56,18 @@ public function __construct(
+     * @param int $ratingLimit
+     * @return $this
+     */
+    public function setRatingLimit($ratingLimit)
```
Can we add strict types to newly added method?
According to the [Technical Guidelines](https://devdocs.magento.com/guides/v2.3/coding-standards/technical-guidelines.html#1-basic-programming-principles) it will be nice to add strict types declaration  for this file.

swnsma commented on app/code/Magento/Sales/Block/Adminhtml/Report/Bestsellers/Filter/Form.php:
```diff
@@ -0,0 +1,40 @@
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Sales\Block\Adminhtml\Report\Bestsellers\Filter;
```
Maybe it is better to place it to Magento\Reports module?

swnsma commented on app/code/Magento/Sales/Block/Adminhtml/Report/Bestsellers/Filter/Form.php:
```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
```
Please add `declare(strict_types=1)`.

swnsma reviewed (changes requested):
Hi @missoxd,

I think we can proceed with the PR as related issue has been approved (so we do not need approve from someone else).

To pass PR for testing we need to have green build.
So please, could you:
1. Check my review comments.
2. Check and fix failed static tests.

sidolov commented:
@missoxd , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!
